### DISCUSSION TITLE
bot does not interact with the windows of the Chest Commands plugin or something like that

### DISCUSSION THREAD

**Discussion by Nekto348** [PrismarineJS/mineflayer#2731] [Q&A] [answered]
I tried to use : bot.moveSlotItem , chest.withdraw , bot.clickWindow , chest.deposit , bot.toss , bot.tossStack. But when using toss (tossStack), there were errors of the type 'the item was not found', and in other cases the code was executed, but nothing happened (the bot remained in the same place, but had to teleport). If who knows how if can fix , please write
Here is my code :

const mineflayer = require('mineflayer')
const { memoryUsage, resourceUsage } = require('process')
const repl = require('repl')
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

const bot = mineflayer.createBot({
  host: 'localhost',
  port: 25565, 
  username: "bot1",
})

bot.on('message', (message) => {
  console.log(message.toAnsi())
 })

bot.on('kicked', (reason, loggedIn) => console.log(reason, loggedIn))
bot.on('error', err => console.log(err))

bot.once('spawn', () => {

console.log('bot1 joined'),
bot.chat ('/l qwertyqwerty'),

setTimeout(function(){
  bot.chat('/murder');
  bot.moveSlotItem(27, 11 );  // this line must take bow from the chest (chest commands  window)
  console.log('mooved')
}2000);

})
  bot.on('windowOpen', () =>
  console.log('window has opened')
  );

bot.on('message', () =>  
  readline.question('you can write', time => {
      
  if (time != ' ' ) {
     bot.chat( time );
  }  
}))

**Comment by amoraschi** [comment, marked as answer]
You could use .clickWindow but you should wait for the window to open

**Reply by Nekto348** [reply]
Thank you very much , it worked
